internal/impl/pure/processor_metric.go

```diff
@@ -299,21 +299,42 @@ func (m *metricProcessor) handleCounterBy(val string, index int, msg message.Bat
 }
 
 func (m *metricProcessor) handleGauge(val string, index int, msg message.Batch) error {
+	var parsedVal any
 	i, err := strconv.ParseInt(val, 10, 64)
-	if err != nil {
-		return err
-	}
-	if i < 0 {
-		return errors.New("value is negative")
+	if err == nil {
+		if i < 0 {
+			return errors.New("value is negative")
+		}
+		parsedVal = i
+	} else {
+		f, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return err
+		}
+		if f < 0 {
+			return errors.New("value is negative")
```

Comment:
Even though it doesn't happen yet elsewhere in this file, it might be nice to include the actual value in the error message, to make troubleshooting easier